packages/playground-examples/copy/ja/TypeScript/Meta-Types/Mapped Types.ts

```diff
@@ -0,0 +1,57 @@
+// mapped typeは他の型を元に新しい型を作る方法であり、
+// 効率的に変換できる型です。
+
+// mapped typeのよくある使い方は
+// 既存の方の部分集合を扱うことです。
+// 例えば、APIが以下のArtistを返すとします:
+
+interface Artist {
+  id: number;
+  name: string;
+  bio: string;
+}
+
+// もし、Artistの一部の変更するAPIを通して
+// 更新を送信したいとき、
+// 一般的には新たに以下の型を作成する必要があるでしょう:
+
+interface ArtistForEdit {
+  id: number;
+  name?: string;
+  bio?: string;
+}
+
+// この型が上記のArtist型と同期されなくなってしまう
+// 可能性は非常に高いでしょう。
+// mapped typeは既存の型を変換した型を作成できます。
+
+type MyPartialType<Type> = {
+  // Typeに含まれるすべてのプロパティを
+  // ?:な値に変換します
+  [Property in keyof Type]?: Type[Property];
+};
+
+// すると、新しくinterfaceを編集した型を作成する代わりに、
+// mapped typeを使用できます。
+type MappedArtistForEdit = MyPartialType<Artist>;
+
+// これでほとんど完璧ですが、
+// これは起こりえないidがnullになるパターンを許容してしまいます。
```

Comment:
[IMO]
`起こりえない` がかかっている箇所がちょっとだけわかりづらかったので、
(一見 `起こりえないidが` とも読めてしまう)
↓のようにするのはいかがでしょうか？

```suggestion
// これは、本来発生しない、idがnullになるパターンを許容してしまいます。
```